k0764427/quiz1.py

```diff
@@ -14,18 +14,26 @@
 #display the value of total followed by a dash and the word total
 
 total = 0
+count = 0
+rule = 0
 
 numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
 
 checkpoint = []
 checkpoint.append("Checkpoint")
 
 for i in numbers: 
+    value = i
     if i % 2 == 0:
-        value = i
         total = total + value
     elif value % 5 == 0:
+            count = count + 1
             checkpoint.append(value)
-print (checkpoint)
+    else:
+        pass
+while rule < count:
```

Comment:
I'd say this is more work than you need to do, but I'm pleased that you can do this correctly too :)

The thing to be wary of with while loops is that it's easy to forget to increment your counter and slip into an infinite loop.

Something like this seems simpler to me

```python
for cp in checkpoint:
    print "Checkpoint %d" % cp
```